test_common.py

```diff
@@ -0,0 +1,219 @@
+    "completeness_score","mutual_info_score","adjusted_mutual_info_score",
+    "normalized_mutual_info_score"
+    ]
+                        
+#metrics which result in 0 when a class is split across different clusters
+ClASS_BASED_METRICS = ["adjusted_mutual_info","normalized_mutual_info",
+    "v_measure_score"
+    ]
+                        
+
+def test_symmetry():
+    #symmetric_metrics
+    for name in SYMMETRIC_METRICS:
+        metric=ALL_METRICS[name]
+        assert_almost_equal(metric([0,1,0],[42,7,42]),
+                            metric([42,7,42], [0,1,0]),
+                            err_msg="%s is not symmetric" % name)
+            
+            
+        assert_almost_equal(metric([0,0,1,1,2,2],[1,1,3,4,2,1]),
+                            metric([1,1,3,4,2,1],[0,0,1,1,2,2]),
+                            err_msg="%s is not symmetric" % name)
+        assert_almost_equal(metric([0,1,2,5,4,9],[0,1,9,4,3,5]),
+                            metric([0,1,9,4,3,5],[0,1,2,5,4,9]),
+                            err_msg="%s is not symmetric" % name)
+                             
+    for name in NON_SYMMETRIC_METRICS:
+        metric=ALL_METRICS[name]
+        assert_almost_equal(metric([0,1,2,5,4,9],[0,1,9,4,3,5]),
+                            metric([0,1,9,4,3,5],[0,1,2,5,4,9]),
+                            err_msg="%s is not symmetric" % name)
+                            
+      
+                    
+#test function for metrics whose output in range 0 to 1
+#bicluster metrics have different input format so they are handled differently                  
+def test_normalized_output_bicluster():
+    # test for jaccard
+    a1 = np.array([True, True, False, False])
+    a2 = np.array([True, True, True, True])
+    a3 = np.array([False, True, True, False])
+    a4 = np.array([False, False, True, True])
+    assert_equal(_jaccard(a1, a1, a1, a1), 1)
+    assert_greater(_jaccard(a1, a1, a2, a2), 0)
+    assert_less(_jaccard(a1, a1, a2, a2), 1)
+    assert_greater(_jaccard(a1, a1, a3, a3),0)
+    assert_less(_jaccard(a1, a1, a3, a3), 1)
+    assert_equal(_jaccard(a1, a1, a4, a4), 0)
+    
+    #test for consensus score 
+    a = [[True, True, False, False],
+        [False, False, True, True]]
+    b = a[::-1]
+    assert_equal(consensus_score((a, a), (a, a)), 1)
+    assert_equal(consensus_score((a, a), (b, b)), 1)
+    assert_equal(consensus_score((a, b), (a, b)), 1)
+    assert_equal(consensus_score((a, b), (b, a)), 1)
+
+    assert_equal(consensus_score((a, a), (b, a)), 0)
+    assert_equal(consensus_score((a, a), (a, b)), 0)
+    assert_equal(consensus_score((b, b), (a, b)), 0)
+    assert_equal(consensus_score((b, b), (b, a)), 0)
+    
+    b=a[: 2]
+    assert_greater(consensus_score((a , a), (b , b)), 0)
+    assert_equal(consensus_score((a , a), (b , b)), 1)
+        
+#test function for metrics whose output in range 0 to 1
+def test_normalized_output():
+    for name in METRICS_NORMALIZED_OUTPUT:

```

Comment:
More generally, we should be able to test that:
* `upper_bound = metric([0, 0, 0, 1, 1, 1], [0, 0, 0, 1, 1, 1]), lower_bound = metric([0, 0, 0, 0, 0, 0], [0, 1, 2, 3, 4, 5])`. Or is this `lower_bound` definition not universally applicable?
* `lower_bound == 0, upper_bound == 1` for normalized metrics
* the cases below fall between the lower and upper bounds, whether normalized or not